x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/signals/open_close_signals_route.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Hapi from 'hapi';
+import { DETECTION_ENGINE_SIGNALS_STATUS_URL } from '../../../../../common/constants';
+import { SignalsRequest } from '../../alerts/types';
+import { setSignalsStatusSchema } from '../schemas';
+import { ServerFacade } from '../../../../types';
+import { transformError, getIndex } from '../utils';
+
+export const setSignalsStatusRouteDef = (server: ServerFacade): Hapi.ServerRoute => {
+  return {
+    method: 'POST',
+    path: DETECTION_ENGINE_SIGNALS_STATUS_URL,
+    options: {
+      tags: ['access:siem'],
+      validate: {
+        options: {
+          abortEarly: false,
+        },
+        payload: setSignalsStatusSchema,
+      },
+    },
+    async handler(request: SignalsRequest, headers) {
+      const { signal_ids: signalIds, query, status } = request.payload;
+      const index = getIndex(request, server);
+      const { callWithRequest } = request.server.plugins.elasticsearch.getCluster('data');
+      let queryObject;
+      if (signalIds) {
+        queryObject = { ids: { values: signalIds } };
+      }
+      if (query) {
+        queryObject = JSON.parse(query);
```

Comment:
I'd just allow an unknown object to be passed in so you don't have to do this parse. Over the wire it has to do a JSON.stringify() and parse anyways before reaching the server as the type will be `application/json`.

Probably easier for the writer of the query and the consumer to have objects even if they are `unknown` to start with.